src/Microsoft.TestPlatform.Utilities/CodeCoverageDataAttachmentsHandler.cs

```diff
@@ -106,15 +112,25 @@ private async Task<string> MergeCodeCoverageFilesAsync(IList<string> files, IPro
 
         private async Task<string> MergeCodeCoverageFilesAsync(IList<string> files, CancellationToken cancellationToken)
         {
-            var coverageUtility = new CoverageFileUtility();
-
-            var coverageData = await coverageUtility.MergeCoverageFilesAsync(
-                    files,
-                    cancellationToken).ConfigureAwait(false);
-
-            coverageUtility.WriteCoverageFile(files[0], coverageData);
-
-            foreach(var file in files.Skip(1))
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var assemblyPath = Path.Combine(Path.GetDirectoryName(typeof(CodeCoverageDataAttachmentsHandler).GetTypeInfo().Assembly.GetAssemblyLocation()), CodeCoverageCoreLibNetAssemblyName + ".dll");
+            
+            // Get assembly, type and methods
```

Comment:
Should we guard against exceptions with a try catch maybe ?

Reply:
It's already cached where this method is called.